var/spack/repos/builtin/packages/visit/package.py

```diff
@@ -62,6 +62,8 @@ class Visit(CMakePackage):
 
     extendable = True
 
+    executables = ['^xml2cmake$']
```

Comment:
This seems like an odd choice for finding a VisIt executable. Is that the only binary available from visit that can print a version?

Reply:
I want 'spack external find visit' since I want VisIt to be extended, which requires that xml2cmake is installed.
But I made a mistake (despite it worked), one should call it with '-version' not '--version'.

Reply:
Hello @sethrj, are you unhappy with my answer, does it nor make sense ?

Reply:
Sorry, I didn't see your earlier response.

It makes sense that `xml2cmake` has to be there to extend visit, but the executable name is so generic that I fear another `xml2cmake` could exist that isn't related to visit. I also anticipate users might be OK with a *non-*extensible version of visit when they find it. So it would make more sense to me for the external to search for the visit executable itself (assuming it can output the version), and the ability to call `xml2cmake` is secondary. *If* the extensible and non-extensible visit installs are different, then you should probably add a variant and use the existence of `xml2cmake` to autodetect the variant.

Maybe @cyrush could comment?

Reply:
Thanks @sethrj for your explanations, I agree with all the issues you draw attention on. I checked the maintainers of packages that depend on visit, maybe @chuckatkins could also tell what he thinks about it from package 'ecp-data-vis-sdk' ?  For package 'sensei' (which depends on libsim which is built deactivating the gui) maintainers are @kwryankrattiger and @sshudler.
About your points @sethrj here is what I know:
- despite `xml2cmake` is generic, I've never seen that outside of VisIt (but who knows in the future),
- for the moment there is no distinction in the VisIt package between extensible or not (ability to build plugins).
It used not to be extensible, and I altered it with VISIT_INSTALL_THIRD_PARTY=ON (thanks to Kathleen Biagas), since I consider a not extensible VisIt is a pity (possibly a shame nowadays). I think that all binary releases of VisIt are now able to build plugins (even on Windows).
- `visit` can output it's version (without igniting the GUI !) - hum, visit is rather generic too, probably too much for anyone else outside the VisIt's team...

All that said, if no one comments, I will put `visit` instead of `xml2cmake` before newt week.

Reply:
As far as I know Visit is currently disabled for both the `sensei` package and the `ecp-data-vis-sdk` at the moment pending fixes in the `visit/package.py`. I am not aware of all of the details on this but for the near term neither package should be affected. @chuckatkins may have more insights on  the reasons for this.

Reply:
Thank you @kwryankrattiger for the feedback. VisIt is a very complex software to build, partly because of it's very open plugin architecture - the reason why I made it "extendable" so that compiling VisIt core does not require building the plugins at the same time (following https://github.com/spack/spack/pull/22907#issuecomment-824218296).
It seems that it is not a sensitive issue, to have external find visit based on `visit` or on `xml2cmake`.
Now, following @sethrj thoughts, I switch to using `visit` since indeed "users might be OK with a *non-*extensible version of visit when they find it".